subsys/nrf_security/Kconfig.legacy

```diff
@@ -670,7 +670,7 @@ endif # MBEDTLS_ECP_C
 config MBEDTLS_HKDF_C
 	bool
 	prompt "HKDF support"
-	default y
+	default y if !TFM_PROFILE_TYPE_MINIMAL
```

Comment:
I am unsure about this change.

I think instead of this maybe revert the changes to nrf_security/cmake/psa_crypto_config.cmake and nrf_security/configs/psa_crypto_config.h.template that was done in this commit: 
https://github.com/nrfconnect/sdk-nrfxlib/commit/457f0de45b55a132b01d9863dc46d5e1f10e5f57